pkg/cable/wireguard/driver.go

```diff
@@ -134,26 +148,38 @@ func (w *wireguard) ConnectToEndpoint(remoteEndpoint types.SubmarinerEndpoint) (
 		return "", fmt.Errorf("failed to parse peer public key: %v", err)
 	}
 
-	klog.V(log.DEBUG).Infof("Connecting cluster %s endpoint %s with publicKey %s", remoteEndpoint.Spec.ClusterID, remoteIP, remoteKey)
+	klog.V(log.DEBUG).Infof("Connecting cluster %s endpoint %s with publicKey %s",
+		remoteEndpoint.Spec.ClusterID, remoteIP, remoteKey)
 	w.mutex.Lock()
 	defer w.mutex.Unlock()
 
-	// delete old peers for ClusterID
+	// delete or update old peers for ClusterID
 	oldKey, found := w.peers[remoteEndpoint.Spec.ClusterID]
 	if found {
 		if oldKey.String() == remoteKey.String() {
-			//TODO check that peer config has not changed (eg allowedIPs)
-			klog.V(log.DEBUG).Infof("Skipping update of existing peer key %s", oldKey)
-			return ip, nil
+			// keys match, existing peer, so lets update status
+			if err := w.updatePeerStatus(remoteKey, remoteEndpoint.Spec.ClusterID); err == nil {
```

Comment:
Why do we update the `Connection` info here as that's done on `GetConnections` anyway?

Reply:
This is not a must, if we assume GetConnections is called periodically.
The status relies on Tx/Rx counts from last check. 
ConnectToEndpoint for existing peers is called periodically so it is an opportunity to refresh Tx/Rx counters.